src/sentry/snuba/metrics/naming_layer/mri.py

```diff
@@ -181,6 +191,46 @@ def mri_string(self) -> str:
         return f"{self.entity}:{self.namespace}/{self.name}@{self.unit}"
 
 
+def is_mri_field(field: str) -> bool:
```

Comment:
I would honestly refactor utilities a bit to centralize MRI parsing here:



```python
@dataclass
class ParsedMRIField:
	op: str
	mri: str

	def __str__() -> str:
		return f"{op}(mri)"

def parse_mri_field(field: str) -> Optional[ParsedMRIField]:
    matches = MRI_EXPRESSION_REGEX.match(field)

    try:
        op = matches[1]
        mri = matches[2]
    except (IndexError, TypeError):
        return None

    return ParseMRIField(aggregate=operation, mri=mri)
```

As a consequence I would implement:

```python
def is_mri_field(field: str) -> bool:
	return parse_mri_field(str) is not None
```

Then the formatting will be as is without the external `parse_mri_field` import.

For that function, I will substitute it to one that takes a `ParsedMRIField` and converts it to `MetricField`.

Reply:
If you would like to bring it to the next level and reduce a lot your parsing of regexes, you could store inside the `ParsedMRIField` directly `ParsedMRI` and directly get the unit from it, so that the formatting method for units would look something like

```python
def format_mri_field_value(field: ParsedMRIField, value: str) -> str:
	unit = field.mri.unit
```

In order to embed `ParsedMRI` into the `ParsedMRIField` you might need to see if the regex you use exposes correctly the nested groups of the embedded `MRI_SCHEMA_REGEX_STRING` regex.

Reply:
Nice! [b7344f9](https://github.com/getsentry/sentry/pull/60859/commits/b7344f9fa9b8f7fa5080afe2b07a55ebd05aea8f)